datacapturing/src/main/java/de/cyface/datacapturing/backend/DataCapturingBackgroundService.kt

```diff
@@ -362,28 +362,28 @@ class DataCapturingBackgroundService : Service(), CapturingProcessListener {
     /**
      * Initializes this service
      *
-     * @param sensorFrequency The frequency in which sensor data should be captured. If this is higher than the maximum
-     * frequency the maximum frequency is used. If this is lower than the maximum frequency the system
-     * usually uses a frequency sightly higher than this value, e.g.: 101-103/s for 100 Hz.
-     * @return the [GeoLocationCapturingProcess]
+     * @param sensorCapture The [SensorCapture] implementation which decides if sensor data should
+     * be captured.
      */
-    private fun initializeCapturingProcess(sensorFrequency: Int): GeoLocationCapturingProcess {
+    private fun initializeCapturingProcess(sensorCapture: SensorCapture):
+            GeoLocationCapturingProcess {
         Log.v(TAG, "Initializing capturing process")
+
         val locationManager = this.getSystemService(LOCATION_SERVICE) as LocationManager
-        Validate.notNull(locationManager)
-        val locationStatusHandler = if (Build.VERSION_CODES.N <= Build.VERSION.SDK_INT
-        ) GnssStatusCallback(locationManager)
-        else GeoLocationStatusListener(locationManager)
-        val sensorManager = getSystemService(SENSOR_SERVICE) as SensorManager
-        Validate.notNull(sensorManager)
-        // noinspection SpellCheckingInspection
-        val geoLocationEventHandlerThread = HandlerThread("de.cyface.locationhandler")
-        // noinspection SpellCheckingInspection
-        val sensorEventHandlerThread = HandlerThread("de.cyface.sensoreventhandler")
-        return GeoLocationCapturingProcess(
-            locationManager, sensorManager, locationStatusHandler,
-            geoLocationEventHandlerThread, sensorEventHandlerThread, sensorFrequency
-        )
+        val locationCapture = LocationCapture()
+        val statusHandler = if (Build.VERSION_CODES.N <= Build.VERSION.SDK_INT) {
+            GnssStatusCallback(locationManager)
+        } else {
+            GeoLocationStatusListener(locationManager)
+        }
+        locationCapture.setup(locationManager, statusHandler)
+
+        if (sensorCapture is SensorCaptureEnabled) {
+            val sensorManager = getSystemService(SENSOR_SERVICE) as SensorManager
+            sensorCapture.setup(sensorManager)
+        }
```

Comment:
This is kind of a Leaky Abstraction, as the BackgroundService should not care about wether sensor capturing has been enabled or not.

Reply:
https://github.com/cyface-de/android-backend/pull/311/commits/975ec75dd2b1ae3755fc4041e219c5ce832a101e